### DISCUSSION TITLE
How to parse booleans?

### DISCUSSION THREAD

**Discussion by camposfyi** [react-querybuilder/react-querybuilder#536] [Q&A] [answered]
I have a Rule type where the input type is `radio`
<img width="884" alt="image" src="https://github.com/react-querybuilder/react-querybuilder/assets/5265784/9357fb1b-6edd-44a4-8972-1a9544202d66">

From the `getValues` function I'm returning the following:
`
return [
        { name: 'true', label: 'true' },
        { name: 'false', label: 'false' },
      ];
`

The value on the rule gets set is 'true' or 'false' which are strings.

I tried handling this during `onQueryChanged`  and converting it to a boolean, but the field `value` in the RuleType is readonly.

I'm using jsonLogic to evaluate my query against an object and it fails because 'true' !== true and 'false' !== false.

**Comment by jakeboone02** [comment, marked as answer]
The read-only thing is probably caused by Immer -- see [here](https://immerjs.github.io/immer/freezing). You can turn it off by running `setAutoFreeze(false)`.

Another option would be to configure the field as `inputType: 'checkbox'` instead, which will set the `value` as an actual `boolean`. If you need the value editor to display radio buttons instead of a checkbox, use a custom value editor like in [this sandbox](https://codesandbox.io/p/sandbox/empty-frog-mc359j?file=%2Fsrc%2FApp.tsx%3A6%2C15):

![image](https://github.com/react-querybuilder/react-querybuilder/assets/366438/5f90bdd7-f0d5-4ad0-a7df-c71c8992b761)

**Comment by camposfyi** [comment]
Thanks Jake. The custom value editor did the ticker.
